libs/app-theme/ui/src/lib/form/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import {
+  ComponentProps,
+  FunctionComponent,
+  InputHTMLAttributes,
+  PropsWithChildren,
+  Ref,
+  useCallback,
+  useEffect,
+  useId,
+  useRef,
+} from "react"
+import styled from "styled-components"
+import {
+  CheckboxSize,
+  IconSize,
+  IconType,
+  TypographyAlign,
+} from "app-theme/models"
+import { Typography } from "../../typography/typography"
+import { Icon } from "../../icon/icon"
+
+interface Props
+  extends PropsWithChildren,
+    Pick<ComponentProps<typeof InputWrapper>, "style" | "className">,
+    Omit<
+      InputHTMLAttributes<HTMLInputElement>,
+      "style" | "className" | "size" | "type" | "placeholder"
+    > {
+  size?: CheckboxSize
+  indeterminate?: boolean
+  ref?: Ref<HTMLInputElement>
+}
+
+export const Checkbox: FunctionComponent<Props> = ({
+  size = CheckboxSize.Large,
+  indeterminate,
+  children,
+  ref,
+  style,
+  className,
+  ...rest
+}) => {
+  const uid = useId()
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const handleRef = useCallback(
```

Comment:
I’ve noticed this small piece of code could be a good candidate for extraction into a UI utility. What do you think?

Reply:
There's even a library for that called [merge-refs](https://www.npmjs.com/package/merge-refs), but I'd leave it for now. Let's see if this will be used more often.